content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
-If the `epoch` attached to the message has a non-reasonable gap (TODO: the gap should be defined) with the routing peer's current `epoch` then the message must be dropped (this is to prevent a newly registered peer spamming the system by messaging for all the past epochs). 
-Furthermore, the routing peers MUST check whether the `proof` is valid and the message is not spam. 
-If both checks are passed successfully, then the message is relayed. 
-If `proof` is invalid then the message is dropped. 
-If spamming is detected, the publishing peer gets slashed. 
-An overview of routing procedure is depicted in Figure 2.
+Proof generation relies on the knowledge of Merkle tree root `merkle_root` and `authPath` which both require access to the membership Merkle tree. 
+Getting access to the Merkle tree can be done in various ways. 
+One way is that all the peers construct the tree locally.
+This can be done by listening to the registration and deletion events emitted by the membership contract.
 
-### Spam Detection and Slashing
-In order to enable local spam detection and slashing, routing peers MUST record the `nullifier`, `share_x`, and `share_y` of any incoming message conditioned that it is not spam and has valid proof. 
-To do so, the peer should follow the following steps. 
-1. The routing peer first verifies the `zkSNARKs` and drops the message if not verified. 
-2. Otherwise, it checks whether a message with an identical `nullifier` has already been relayed. 
-   1. If such message exists and its `share_x` and `share_y` components are different from the incoming message, then slashing takes place (if the `share_x` and `share_y` fields of the previously relayed message is identical to the incoming message, then the message is a duplicate and shall be dropped).
-   2. If none found, then the message gets relayed.
+For the group synchronization, one important security consideration is that peers MUST make sure they always use the most recent Merkle tree root in their proof generation.
+The reason is that using an old root can allow inference about the index of the user's `pk` in the membership tree hence compromising user privacy and breaking message unlinkability.
 
-An overview of slashing procedure is provided in Figure 2.
+## Routing
 
-<!-- TODO: may shorten or delete the Spam detection and slashing process -->
+Upon the receipt of a PubSub message via [`11/WAKU2-RELAY`](/spec/11) protocol, the routing peer parses the `data` field as a `WakuMessage` and gets access to the `RateLimitProof` field.  
+The peer then validates the `RateLimitProof`  as explained next.
 
-<!-- TODO: may consider [validator functions](https://github.com/libp2p/specs/tree/master/pubsub#topic-validation) or [extended validators](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators) for the spam detection -->
+**Epoch Validation**
```

Comment:
### perhaps?

Reply:
epoch validation is already discussed a few paragraphs above, are you suggesting to move it to here?

Reply:
I actually forgot what this was about, so I guess we can ignore it :D